urlfcc351a1ba_kuntapavankumar_Z_tgz-pJ8-edu_concordia_dpis_ReplicaTestJ8/src/edu/concordia/dpis/MulticastServer.java

```diff
@@ -0,0 +1,86 @@
+package edu.concordia.dpis;
+
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+
+@Deprecated
+public class MulticastServer extends Thread {
+	
+	private long FIVE_SECONDS = 5000;
+	private int itHasMoreRequest = 1;
+	private int port;
+	
+	
+	public MulticastServer(int port){
+		this.port = port;	
+	}
+
+	public void run(){
+		
+		do
+		{
+			DatagramSocket socket = null;
+			try
+			{	System.out.println("server is up");
+				socket = new DatagramSocket();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *socket*. Execution paths that do not contain closure statements were detected. Ensure *socket* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).